homeassistant/components/binary_sensor/__init__.py

```diff
@@ -150,13 +151,15 @@ class BinarySensorEntity(Entity):
     """Represent a binary sensor."""
 
     _attr_is_on: bool | None = None
+    _attr_state: None = None
 
     @property
     def is_on(self) -> bool | None:
         """Return true if the binary sensor is on."""
         return self._attr_is_on
 
     @property
+    @final
```

Comment:
We normally put this on top of the `property` decorator.

Reply:
I find one occurrence in the cover integration where we do the opposite.

Reply:
![image](https://user-images.githubusercontent.com/195327/120467702-ba236e80-c3a0-11eb-9918-898bfd6aedb0.png)

🤷‍♂️

Reply:
What linter is that?

Reply:
Pylance/pyright (VSCode)

Reply:
I can't find any examples of our specific application in standard library docs or the related PEP. Seems weird to not be able to put the final decorator on top.

Reply:
I'm fine with it either way.
I was just annoyed by the warning in my face, hence I swapped it. IMHO it's a bug in pyright.